### PR TITLE
fix(web): fixes touch form-factor default kbd on cookieless keymanweb.com page load

### PR DESCRIPTION
**Pull request by jahorton**
Fixes https://github.com/keymanapp/keymanweb.com/issues/86.

Refer to #9304, which provides lots of very relevant context for the issue and the needed fix.

Unfortunately, the main way I know to test this fix... is to wait for the next alpha build and try then. 

It should be possible to build a test page with a similar rigging when time permits, come to think of it, but no such test page exists yet.

@keymanapp-test-bot skip

### CODE REVIEW

File: web/src/app/browser/src/keymanEngine.ts [hunk 1]
```diff
@@ -175,6 +175,19 @@ export default class KeymanEngine extends KeymanEngineBase<BrowserConfiguration,
 
     await super.init(totalOptions);
 
+    // Used by keymanweb.com; if no keyboard-cookie exists, we need this to trigger
```
Comment (mcdurdin) [inline web/src/app/browser/src/keymanEngine.ts:178]: Really only used by keymanweb.com? Not by almost every single site that uses keymanweb? After all, no sites will have cookies on initial use.

**Review by mcdurdin — APPROVED** [review]
I don't really understand the fix -- it seems complex. But LGTM

File: web/src/app/browser/src/keymanEngine.ts (hunk 1 above)
Comment (jahorton) [reply web/src/app/browser/src/keymanEngine.ts:178]: Most sites don't directly call `keyman.register`.

keymanweb.com's `cdn/prog/keyboards.php` resolves to this due to caching:

```javascript
/* tier alpha, version 17.0, cached 2023-10-18T02:25:03+00:00 */
function addKeyboards() {
    keyman.register({
        "options": {
// ...
```

which is then called here:

https://github.com/keymanapp/keymanweb.com/blob/0dba6472d3300ef15a4820ee53953d0051333aa0/inc/head.php#L207-L215

```javascript
  keyman.init({
    attachType:'auto',
    setActiveOnRegister: setActiveOnRegister
  }).then(function() {
    if(typeof afterInit == 'function') {
      afterInit();
    }
    if(typeof addKeyboards == 'function') {
      addKeyboards();
```

The site completely bypasses KeymanWeb's `addKeyboards` API, using an unofficial, undocumented optimization.  Therefore...

> not by almost every single site that uses keymanweb

File: web/src/app/browser/src/keymanEngine.ts [hunk 2]
```diff
@@ -175,6 +175,19 @@ export default class KeymanEngine extends KeymanEngineBase<BrowserConfiguration,
...
+    // default-stub selection on mobile devices so that a keyboard - and thus, the
+    // globe key - are accessible.
+    //
+    // The `super` call above initializes `keyboardRequisitioner`, as needed here.
+    this.keyboardRequisitioner.cloudQueryEngine.once('unboundregister', () => {
```
Comment (jahorton) [inline web/src/app/browser/src/keymanEngine.ts:183]: This event was added in #9304 to facilitate handling of keymanweb.com's direct call of `keyman.register`.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.194-alpha](https://keyman.com/downloads/releases/alpha/17.0.194)

**Pull request merged**